src/bin/purge_ttl.rs

```diff
@@ -298,34 +308,69 @@ fn main() -> Result<(), Box<dyn Error>> {
         let _timer_total = start_timer(&statsd, "purge_ttl.total_duration");
         {
             let _timer_batches = start_timer(&statsd, "purge_ttl.batches_duration");
-
-            if incremental {
-                delete_incremental(
-                    &client,
-                    &session,
-                    "batches".to_owned(),
-                    "batch_id".to_owned(),
-                    chunk_size,
-                    max_to_delete,
-                )?;
-            } else {
-                delete_all(&client, &session, "batches".to_owned())?;
+            let mut success = false;
+            for i in 0..retries {
+                match if incremental {
+                    delete_incremental(
+                        &client,
+                        &session,
+                        "batches".to_owned(),
+                        "batch_id".to_owned(),
+                        chunk_size,
+                        max_to_delete,
+                    )
+                } else {
+                    delete_all(&client, &session, "batches".to_owned())
+                } {
+                    Ok(_) => {
+                        success = true;
+                        break;
+                    }
+                    Err(e) => {
```

Comment:
I think this should check against specific error codes. It looks like everything is currently `dyn Error` so we'd have to start returning grpcio Errors to do this.